saleor/discount/utils.py

```diff
@@ -1024,39 +1096,43 @@ def get_variants_to_promotion_rules_map(
     return rules_info_per_variant
 
 
-def fetch_promotion_rules_for_checkout(
-    checkout: Checkout,
+def fetch_promotion_rules_for_checkout_or_order(
+    instance: Union["Checkout", "Order"],
     database_connection_name: str = settings.DATABASE_CONNECTION_DEFAULT_NAME,
 ):
     from ..graphql.discount.utils import PredicateObjectType, filter_qs_by_predicate
 
     applicable_rules = []
     promotions = Promotion.objects.active()
-    checkout_channel_id = checkout.channel_id
-    PromotionRuleChannels = PromotionRule.channels.through.objects.filter(
-        channel_id=checkout_channel_id
-    )
     rules = (
         PromotionRule.objects.using(database_connection_name)
-        .filter(
-            Exists(promotions.filter(id=OuterRef("promotion_id"))),
-            Exists(PromotionRuleChannels.filter(promotionrule_id=OuterRef("id"))),
-        )
+        .filter(Exists(promotions.filter(id=OuterRef("promotion_id"))))
         .exclude(order_predicate={})
+        .prefetch_related("channels")
     )
+    rule_to_channel_ids_map = _get_rule_to_channel_ids_map(rules)
 
-    currency = checkout.currency
-    checkout_qs = Checkout.objects.using(database_connection_name).filter(
-        pk=checkout.pk
+    channel_id = instance.channel_id
+    currency = instance.channel.currency_code
+    qs = instance._meta.model.objects.using(database_connection_name).filter(  # type: ignore[attr-defined] # noqa: E501
```

Comment:
Why not:
```suggestion
    qs = instance.objects.using(database_connection_name).filter(
```
I might already asked about this somewhere 😅

Reply:
Manager is not accessible via model instance